crates/sui/src/upgrade_compatibility/mod.rs

```diff
@@ -1051,6 +1051,19 @@ fn missing_definition_diag(
 ) -> Result<Diagnostics, Error> {
     let mut diags = Diagnostics::new();
 
+    // capitalize the first letter
+    let capital_declaration_kind = declaration_kind
+        .chars()
+        .enumerate()
+        .map(|(i, c)| {
+            if i == 0 {
+                c.to_uppercase().next().unwrap_or(c)
+            } else {
+                c
+            }
+        })
+        .collect::<String>();
```

Comment:
nit: here's a more robust `capitalize` function:

```rust
fn capitalize(s: &str) -> String {
    let mut cs = s.chars();
    let mut capitalized = String::new();
    let Some(fst) = cs.next() else {
        return capitalized;
    };
    
    capitalized.extend(fst.to_uppercase());
    capitalized.extend(cs);
    capitalized
}
```

The difference between these two is only going to be relevant if the first character's uppercase character could extend beyond one `char`, which seems unlikely in this case, but still.